internal/pkg/install/install.go

```diff
@@ -0,0 +1,228 @@
+					return err
+				}
+				destFile, err = os.Create(filepath.Join(dest, filepath.Base(artifact)))
+				if err != nil {
+					return err
+				}
+			default:
+				// Local directories/links
+				link := strings.Split(artifact, ":")
+				if len(link) == 1 {
+					if err = os.MkdirAll(filepath.Join(dest, artifact), 0755); err != nil {
+						return err
+					}
+				} else {
+					if err = os.Symlink(link[1], filepath.Join(dest, link[0])); err != nil && !os.IsExist(err) {
+						return err
+					}
+				}
+			}
+
+			switch {
+			case strings.HasSuffix(sourceFile.Name(), ".tar") || strings.HasSuffix(sourceFile.Name(), ".tar.gz"):
+				log.Printf("Extracting %s\n", artifact)
+				err = untar(sourceFile, dest)
+				if err != nil {
+					return err
+				}
+			case strings.HasPrefix(sourceFile.Name(), "/"):
+				if _, err = io.Copy(destFile, sourceFile); err != nil {
+					return err
+				}
+
+				if err = destFile.Close(); err != nil {
+					return err
+				}
+			default:
+			}
+
+			if err = sourceFile.Close(); err != nil {
+				return err
+			}
+			if err = os.Remove(sourceFile.Name()); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// Simple extract function
+// nolint: gocyclo, dupl
+func untar(tarball *os.File, dst string) error {
+
+	var input io.Reader
+	var err error
+
+	if strings.HasSuffix(tarball.Name(), ".tar.gz") {
+		input, err = gzip.NewReader(tarball)
+		if err != nil {
+			return err
+		}
+
+		// nolint: errcheck
+		defer input.(*gzip.Reader).Close()
+	} else {
+		input = tarball
+	}
+
+	tr := tar.NewReader(input)
+
+	for {
+		var header *tar.Header
+
+		header, err = tr.Next()
+
+		switch {
+		case err == io.EOF:
+			err = nil
+			return err
+		case err != nil:
+			return err
+		case header == nil:
+			continue
+		}
+
+		// the target location where the dir/file should be created
+		target := filepath.Join(dst, header.Name)
+
+		// May need to add in support for these
+		/*
+			// Type '1' to '6' are header-only flags and may not have a data body.
+				TypeLink    = '1' // Hard link
+				TypeSymlink = '2' // Symbolic link
+				TypeChar    = '3' // Character device node
+				TypeBlock   = '4' // Block device node
+				TypeDir     = '5' // Directory
+				TypeFifo    = '6' // FIFO node
+		*/
+		switch header.Typeflag {
+		case tar.TypeDir:
+			if err = os.MkdirAll(target, 0755); err != nil {
+				return err
+			}
+		case tar.TypeReg:
+			var downloadedFileput *os.File
+
+			downloadedFileput, err = os.OpenFile(target, os.O_CREATE|os.O_RDWR, os.FileMode(header.Mode))
+			if err != nil {
+				return err
+			}
+
+			if _, err = io.Copy(downloadedFileput, tr); err != nil {
+				return err
+			}
+
+			err = downloadedFileput.Close()
+			if err != nil {
+				return err
+			}
+		case tar.TypeSymlink:
+			dest := filepath.Join(dst, header.Name)
+			source := header.Linkname
+			if err := os.Symlink(source, dest); err != nil {
+				return err
+			}
+		}
+	}
+}
+
+func downloader(artifact *url.URL, base string) (*os.File, error) {
```

Comment:
```suggestion
func download(artifact *url.URL, base string) (*os.File, error) {
```
`downloader`, to me at least, makes it sound like it returns a struct/interface of something that downloads ... stuff :). Its a nit, but `download` sounds more appropriate here IMHO.